### PR TITLE
pull in latest FMU-related libraries from boptest

### PR DESCRIPTION
**Pull request by nllong**
This is a simple update to pull in the data_generator.py file from BOPTEST's repo. The Dockerfile and step_run update are small changes not related to the pull from BOPTEST.

### CODE REVIEW

File: alfalfa_worker/jobs/modelica/step_run.py [hunk 1]
```diff
@@ -51,7 +51,6 @@ def __init__(self, run_id, realtime, timescale, external_clock, start_datetime="
         self.tc = TestCase(**config)
 
         # run the FMU simulation
-        self.kstep = 0  # todo remove if not used
```
Comment (nllong) [inline alfalfa_worker/jobs/modelica/step_run.py:54]: says "remove if not used". I don't see it being used :)

File: alfalfa_worker/lib/data/data_generator.py [hunk 2]
```diff
@@ -164,17 +162,15 @@ def generate_weather(self,
...
         with open(model_file) as f:
             newText = f.read().replace(str_old, str_new)
 
         with open(model_file, "w") as f:
             f.write(newText)
 
         # Change to Resources directory
-        currdir = os.getcwd()
+        currdir = os.curdir
         os.chdir(self.weather_dir)
```
Comment (TShapinsky) [inline alfalfa_worker/lib/data/data_generator.py:173]: Is there a way we can do this stuff without changing dir?

Comment (nllong) [reply alfalfa_worker/lib/data/data_generator.py:173]: I'm sure there is, but I really don't want to touch this code until we can get adequate testing implemented on this file. I actually plan on removing `data_generator` for our [`fmu_runner.py`](https://github.com/urbanopt/geojson-modelica-translator/blob/develop/geojson_modelica_translator/modelica/lib/runner/fmu_runner.py), eventually.

Comment (TShapinsky) [reply alfalfa_worker/lib/data/data_generator.py:173]: Makes sense. I currently have the dispatcher reset the CWD every time it gets a new job (due to MLEP changing directory) so this should be fine

**Review by TShapinsky — APPROVED** [review]

**Pull request merged**